### PR TITLE
Refactor Matrix4::ToQuaternion()

### PR DESCRIPTION
**Pull request by jonschz**
This implementation of `Matrix4::ToQuaternion()` matches up to the stack layout of the function. Not sure if this is compiler entropy or if this can be influenced in the source code.

### CODE REVIEW

File: LEGO1/realtime/matrix.h [hunk 1]
```diff
@@ -151,48 +151,47 @@ class Matrix4 {
...
-	float valueB = *Element(next, next);
-	float valueC = *Element(largest, largest);
 
-	// Above is somewhat decomped, below is pure speculation since the automatic
-	// decomp becomes very garbled.
-	float traceValue = sqrt(valueA - valueB - valueC + 1.0);
+	trace = *Element(nextNext, nextNext);
+	trace += *Element(next, next);
+	trace = *Element(largest, largest) - trace;
+	trace += 1.0f;
```
[the remark this answers is not in the document]
Comment (jonschz) [reply LEGO1/realtime/matrix.h:186]: Really? `trace` is declared as a `float` and so is the return type of `Element`. I was under the impression that `1.0f` is a `float` while `1.0` is a `double`.

File: LEGO1/realtime/matrix.h [hunk 2]
```diff
@@ -151,48 +151,47 @@ class Matrix4 {
...
+	trace = sqrt(trace);
```
[the remark this answers is not in the document]
Comment (jonschz) [reply LEGO1/realtime/matrix.h]: good catch! I did try the new variable once before, but back then it did not improve anything due to other errors still present.

**Comment by jonschz** [comment]
I tried for a bit and I can't get the 100 % easily. It seems like the recompile always calls `*Element(next, next)` before `*Element(nextNext, nextNext)` no matter in what order I put in the arguments.

**Comment by foxtacles** [comment]
According to the most recent reccmp diff in this PR (which includes changes from master) the function ends up 100%:

```
Matrix4::ToQuaternion (0x10002550) is a stub.
Matrix4::ToQuaternion (0x10002550) is 100.00% similar to the original
```

So it's relatively safe to assume that any changes to that figure from now on is just entropy

**Pull request merged**